validator/sawtooth_validator/execution/scheduler_serial.py

```diff
@@ -49,6 +49,9 @@ def __init__(self, squash_handler, first_state_hash):
         self._last_in_batch = []
         self._last_state_hash = first_state_hash
 
+    def __del__(self):
```

Comment:
Ah, cool. I was getting __del__ confused with __delitem__. The scheduler contains no external resources so doesn't have to release anything upon garbage collection. In the current code, the scheduler won't be garbage collected until either finalize or cancel is called, so calling .cancel again on garbage collection doesn't make much sense.